components/org.wso2.carbon.identity.api.server.identity.governance/org.wso2.carbon.identity.api.server.identity.governance.v1/src/main/java/org/wso2/carbon/identity/api/server/identity/governance/v1/core/ServerIdentityGovernanceService.java

```diff
@@ -159,7 +163,8 @@ public ConnectorRes getGovernanceConnector(String categoryId, String connectorId
                     .withoutPadding()
                     .encodeToString(connectorConfig.getCategory().getBytes(StandardCharsets.UTF_8));
             if (!categoryId.equals(categoryIdFound)) {
-                throw handleNotFoundError(connectorId, GovernanceConstants.ErrorMessage.ERROR_CODE_CONNECTOR_NOT_FOUND);
+                throw handleMismatchError(connectorId, categoryId,
```

Comment:
By modifying this are we changing the response as well? If so there will be backward compatibility issues.

Reply:
The response status code is still the same. It's a `404 - Not Found`. Only the error message in the response is changed. 

Would keeping this change be okay or shall I revert it to the previous one?

Reply:
As this is not a new API call we can not change the response of an existing API call as it leads backward compatibility issues.

Reply:
Ack